src/components/CommentBox/CommentBox.tsx

```diff
@@ -17,25 +29,61 @@ const CommentBox = () => {
     }
   }, [comment]);
 
-  const handleComment = () => {
-    if (comment && comment.length > 0) {
-      submitComment(true);
+  useEffect(() => {
+    if (content) {
+      setInputModeActive(false);
+      setComment(content);
     }
+  }, []);
+
+  const handleComment = () => {
+    if (!comment || comment.length === 0)
+      return dispatch(deleteCommentAction(id));
+
+    setInputModeActive(false);
+    dispatch(editCommentContentAction({ id, content: comment }));
+    dispatch(handleCommentVisibilityAction({ id, isOpen: false }));
   };
 
   return (
     <div className={styles.wrapper}>
-      <div className={styles.headerIcon}>
-        <UserIcon />
+      <div className={styles.header}>
+        <div className={styles.headerContent}>
+          <UserIcon />
+          User
+        </div>
+        <div className={styles.iconSection}>
+          <div
```

Comment:
suggestion: replace divs with "onClick" with `<button type="button">` for better accessibility and semantics.